components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -71,6 +71,29 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
         break
       }
 
+      if (tab && tab.url && tab.url.includes('twitch.tv')) {
+        let url = tab.url
+        // need a delay here to allow the DOM to finish generating
+        setTimeout(function () {
```

Comment:
can we detect when dom has loaded? this approach can be tricky on slow connections and user needs to wait 10s before they can use panel

Reply:
DOM is constantly regenerating while user is on the page so I'm not sure we would get any reliable detection that DOM is 'done'. I'm thinking of parsing the DOM in this area and setting an interval when we get the name and favicon.

Reply:
my concern is that this is not bullet proof and we will get a lot of reports that something is not working, because of 10s delay or not loading in 10s

Reply:
I'll definitely get with you to adjust and stabilize as needed.